src/test/ui/impl-trait/static-return-lifetime-infered.stderr

```diff
@@ -1,33 +1,33 @@
-error: cannot infer an appropriate lifetime
+error[E0758]: cannot infer an appropriate lifetime
   --> $DIR/static-return-lifetime-infered.rs:7:16
    |
 LL |     fn iter_values_anon(&self) -> impl Iterator<Item=u32> {
-   |                         -----     ----------------------- ...is required to be `'static` by this...
+   |                         -----     ----------------------- ...is required to live as long as `'static` by this...
```

Comment:
Perhaps `by…` instead, as now we have two `this`s.

Reply:
What about 
```
...is required to live as long as `'static` here...
```